dxr/build.py

```diff
@@ -523,6 +522,11 @@ def docs():
                     total['annotations'] = annotations_for_this_line
                 yield es.index_op(total)
 
+                # Because needles_by_line holds a reference, total is not
```

Comment:
I like the clearing better because it's noticeably faster (5 seconds across an 80-second index—correct me if I'm wrong). Let's just document the hell out of it.